cpp/src/link_prediction/jaccard_impl.cuh

```diff
@@ -0,0 +1,72 @@
+/*
+ * Copyright (c) 2022, NVIDIA CORPORATION.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+#pragma once
+
+#include <cugraph/algorithms.hpp>
+#include <link_prediction/similarity_impl.cuh>
+
+#include <raft/handle.hpp>
+
+namespace cugraph {
+namespace detail {
+
+struct jaccard_functor_t {
+  template <typename weight_t>
```

Comment:
Why's this placed after struct?

Reply:
I was templating the method rather than the structure since I'm not defining any member data.

This allows me to pass the struct through different functions without needing to compile multiple variations, the multiple variations are only compiled in the place where the function is actually called.

I can change this if we really think it would be better on the struct itself.  I don't expect that I will actually pass the struct through a bunch of function calls when I finish the implementation.

Reply:
I see. Yeah, not a big deal, I just thought this was a mistake, but I now learned this is intended.